packages/style-engine/class-wp-style-engine-css-rules-store.php

```diff
@@ -109,30 +109,34 @@ public function get_all_rules() {
 		 * Gets a WP_Style_Engine_CSS_Rule object by its selector.
 		 * If the rule does not exist, it will be created.
 		 *
-		 * @param string $selector The CSS selector.
-		 * @param string $at_rule  The CSS nested @rule, such as `@media (min-width: 80rem)` or `@layer module`.
+		 * @param string|WP_Style_Engine_CSS_Rule|WP_Style_Engine_CSS_Rules_Container $rule The CSS selector or a WP_Style_Engine_CSS_Rule|WP_Style_Engine_CSS_Rules_Container object.
 		 *
 		 * @return WP_Style_Engine_CSS_Rule|void Returns a WP_Style_Engine_CSS_Rule object, or null if the selector is empty.
 		 */
-		public function add_rule( $selector, $at_rule = '' ) {
-			$selector = trim( $selector );
-			$at_rule  = trim( $at_rule );
-
-			// Bail early if there is no selector.
-			if ( empty( $selector ) ) {
+		public function add_rule( $rule ) {
+			if ( empty( $rule ) ) {
 				return;
 			}
 
-			if ( ! empty( $at_rule ) ) {
-				if ( empty( $this->rules[ "$at_rule $selector" ] ) ) {
-					$this->rules[ "$at_rule $selector" ] = new WP_Style_Engine_CSS_Rule( $selector, array(), $at_rule );
-				}
-				return $this->rules[ "$at_rule $selector" ];
+			$is_rules_object = $rule instanceof WP_Style_Engine_CSS_Rules_Container || $rule instanceof WP_Style_Engine_CSS_Rule;
+
+			if ( $is_rules_object ) {
+				$selector = $rule->get_selector();
+			}
+
+			if ( is_string( $rule ) ) {
+				$selector = trim( $rule );
+			}
+
+			if ( empty( $selector ) ) {
+				return;
 			}
 
-			// Create the rule if it doesn't exist.
+			/*
+				Create a new WP_Style_Engine_CSS_Rule rule by default if it doesn't exist.
+			*/
 			if ( empty( $this->rules[ $selector ] ) ) {
-				$this->rules[ $selector ] = new WP_Style_Engine_CSS_Rule( $selector );
+				$this->rules[ $selector ] = $is_rules_object ? $rule : new WP_Style_Engine_CSS_Rule( $selector );
```

Comment:
If the selector is a string create a CSS_Rule. This is legacy behaviour. Containers, for now should be passed as already instantiated.